src/main/java/kg/jarkyn/cobspecserver/Runner.java

```diff
@@ -0,0 +1,32 @@
+package kg.jarkyn.cobspecserver;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class Runner {
+    public static void main(String[] args) {
+        server().run();
+    }
+
+    private static Server server() {
+        return new Server(listener(), controller());
+    }
+
+    private static ListenerSocket listener() {
+        return new ListenerSocket(getServerSocket(), true);
```

Comment:
I feel like this is a place that needs a constant to add meaning. I can't tell what `true` means in this context without opening up the `ListenerSocket` class. You could add a constant `KEEP_RUNNING = true` to express more clearly what the boolean flag means.